spec/requests/apartments_spec.rb

```diff
@@ -1,5 +1,91 @@
 require 'rails_helper'
 RSpec.describe "Apartments", type: :request do
+
+  describe 'PATCHES /update' do
+    it 'can update an apartment with a specific ID' do
+
+      user = User.where(email: 'test@test.test').first_or_create(password: '12345678', password_confirmation: '12345678')
+      
+      user.apartments.create(
+          street: "123 Main St",
+          city:"San Diego",
+          state:"CA",
+          manager:"Mr. Magoo",
+          email:"magoo@example.com", 
+          price:"2k", 
+          bedrooms:2, 
+          bathrooms:2, 
+          pets:"cats only",
+          image:"exampleimage.com", 
+          user_id: user.id
+      )      
+        
+
+      apartment_params = {
+        apartment: {
+          street: "6778 Hill St",
+          city:"San Francisco",
+          state:"CA",
+          manager:"Mr. Hello",
+          email:"hello@example.com", 
+          price:"2k", 
+          bedrooms:2, 
+          bathrooms:2, 
+          pets:"cats only",
+          image:"exampleimage.com", 
+          user_id: user.id        }
+      }
+
+      apartment = Apartment.last
+      patch "/apartments/#{apartment.id}", params: apartment_params
+      expect(response).to have_http_status(200)
+      updated_apartment = Apartment.find(apartment.id)
+
+      expect(updated_apartment.street).to eq "6778 Hill St"
+      expect(updated_apartment.city).to eq "San Francisco"
+    end
+      it "doesn't update an apartment without a street, city, state, manager, etc" do
+        user = User.where(email: 'test@test.test').first_or_create(password: '12345678', password_confirmation: '12345678')
+        
+        user.apartments.create(
+            street: "123 Main St",
+            city:"San Diego",
+            state:"CA",
+            manager:"Mr. Magoo",
+            email:"magoo@example.com", 
+            price:"2k", 
+            bedrooms:2, 
+            bathrooms:2, 
+            pets:"cats only",
+            image:"exampleimage.com", 
+            user_id: user.id
+        )      
+        apartment_params = {
+          apartment: {               
+            street: "",
+            city:"",
+            state:"",
+            manager:"",
+            email:"magoo@example.com", 
+            price:"2k", 
+            bedrooms:2, 
+            bathrooms:2, 
+            pets:"cats only",
+            image:"exampleimage.com", 
+            user_id: user.id    }
+        }
+        apartment = Apartment.last
+        patch "/apartments/#{apartment.id}", params: apartment_params
+        expect(response.status).to eq(422)
+        json = JSON.parse(response.body)
+        expect(json['street']).to include "can't be blank"
+        expect(json['city']).to include "can't be blank"
+        expect(json['state']).to include "can't be blank"
+        expect(json['manager']).to include "can't be blank"
```

Comment:
You are already here, just add in the other fields.

Reply:
ok will do